### ISSUE TITLE
SamlAuthService fetchUserRecord return always null if pid not 0

### ISSUE THREAD

**Issue by Bruno86** [cdaecke/md_saml#16]
Hello,
In SamlAuthService the method fetchUserRecord return always null if the pid != 0
this code have no effect : 
```
        if ($loginType === 'FE' && isset($extSettings['fe_users']['databaseDefaults']['pid'])) {
            $pid = (int)$extSettings['fe_users']['databaseDefaults']['pid'];
            $this->db_user['check_pid_clause'] = '`pid` IN (' . $pid . ')';
        }
```

PS : Typo3 v12.4.10

**Comment by cdaecke** [comment]
Please use [version 2 ](https://github.com/cdaecke/md_saml/releases/tag/2.0.0)of the extension for the moment. Version 3 for TYPO3 v12 is not working correctly up to now.

Any suggestions and pull requests are welcome.

**Comment by AlexKvrlp** [comment]
I added a pull request to fix this. In the meantime you could use my gist. Just a it to your sitepackage.
https://gist.github.com/AlexKvrlp/48cce2dd7d137bcb2333bf29f5b1f77e

**Comment by Bruno86** [comment]
Your solution works fine and better than mine.  Thank you :)

**Comment by cdaecke** [comment]
This was solved with #25
